src/command/install/plugin.rs

```diff
@@ -32,21 +32,34 @@ impl Plugin {
     pub fn get_tarball_url(&self) -> Result<String> {
         match self {
             Self::Supergraph(v) => {
-                let target_arch = if cfg!(target_os = "windows") {
-                    Ok("x86_64-pc-windows-msvc")
+                let mut target_arch = Err(anyhow!(
+                    "Your current architecture does not support installation of this plugin."
+                ));
+                if cfg!(target_os = "windows") {
+                    target_arch = Ok("x86_64-pc-windows-msvc");
                 } else if cfg!(target_os = "macos") {
-                    Ok("x86_64-apple-darwin")
+                    // we didn't always build aarch64 MacOS binaries,
+                    // so check to see if this version supports them or not
+                    if v.supports_arm() && cfg!(target_arch = "aarch64") {
+                        target_arch = Ok("aarch64-apple-darwin");
+                    } else {
+                        // if it isn't supported, download the x86_64 version
+                        // since Rosetta will make sure it works
+                        target_arch = Ok("x86_64-apple-darwin")
+                    }
+                // unfortunately, deno does not support musl architectures
+                // so we do not download the supergraph plugin on those machines
                 } else if cfg!(target_os = "linux") && !cfg!(target_env = "musl") {
-                    Ok("x86_64-unknown-linux-gnu")
-                } else {
-                    Err(anyhow!(
-                        "Your current architecture does not support installation of this plugin."
-                    ))
-                }?;
+                    if cfg!(target_arch = "x86_64") {
+                        target_arch = Ok("x86_64-unknown-linux-gnu");
+                    } else if cfg!(target_arch = "aarch64") {
+                        target_arch = Ok("aarch64-unknown-linux-gnu");
+                    }
+                }
```

Comment:
i d have probably gone with match (target_os, target_env) here, but i m not even sure it would have looked better 😅